src/g2getgb2.F90

```diff
@@ -570,14 +590,111 @@ end subroutine getgb2rp
 
   ! Extract grib message from file.
   nullify(gribm)
-  call getgb2rp(lugb, cbuf(lpos:), extract, gribm, leng, iret)
+  call getgb2rp2(lugb, idxver, cbuf(lpos:), extract, gribm, leng, iret)
 
   k = jk
 
-  ! Free cbuf memory allocated in getg2i/getg2ir().
+  ! Free cbuf memory allocated in getg2i/getg2i2r().
   if (associated(cbuf)) deallocate(cbuf)
   
   call gf_free(gfld)
+end subroutine getgb2p2
+
+!> Find and extract a GRIB2 message from a file, using index version
+!> 1. Use getgb2p2() to use either index version 1 or version 2.
```

Comment:
I'm a little confused by the documentation-- Does this subroutine do just index version 1 or does it handle both?